### PR TITLE
Trainer constructor cleanup

### PR DESCRIPTION
**Pull request by shaydeci**

### CODE REVIEW

File: src/super_gradients/training/sg_trainer/sg_trainer.py [hunk 1]
```diff
@@ -76,30 +74,20 @@ class Trainer:
...
-                 train_loader: DataLoader = None, valid_loader: DataLoader = None, test_loader: DataLoader = None,
-                 classes: List[Any] = None):
+    def __init__(self, experiment_name: str, device: str = None,
+                 multi_gpu: Union[MultiGPUMode, str] = MultiGPUMode.OFF, ckpt_name: str = 'ckpt_latest.pth',
+                 ckpt_root_dir: str = None):
         """
 
         :param experiment_name:                      Used for logging and loading purposes
         :param device:                          If equal to 'cpu' runs on the CPU otherwise on GPU
         :param multi_gpu:                       If True, runs on all available devices
```
Comment (ofrimasad) [inline src/super_gradients/training/sg_trainer/sg_trainer.py:84]: can't we move multi gpu mode to training params as well?

File: src/super_gradients/training/sg_trainer/sg_trainer.py [hunk 2]
```diff
@@ -76,30 +74,20 @@ class Trainer:
...
-        :param model_checkpoints_location:      If set to 's3' saves the Checkpoints in AWS S3
                                                 otherwise saves the Checkpoints Locally
-        :param overwrite_local_checkpoint:      If set to False keeps the current local checkpoint when importing
                                                 checkpoint from cloud service, otherwise overwrites the local checkpoints file
         :param ckpt_name:                       The Checkpoint to Load
```
Comment (ofrimasad) [inline src/super_gradients/training/sg_trainer/sg_trainer.py:87]: why do we need ckpt_name here?

File: src/super_gradients/training/sg_trainer/sg_trainer.py [hunk 3]
```diff
@@ -859,6 +834,7 @@ def forward(self, inputs, targets):
 
         self.train_loader = train_loader or self.train_loader
         self.valid_loader = valid_loader or self.valid_loader
+        self._set_dataset_params()
```
Comment (ofrimasad) [inline src/super_gradients/training/sg_trainer/sg_trainer.py:837]: on each backward step? why?

Comment (ofrimasad) [reply src/super_gradients/training/sg_trainer/sg_trainer.py:837]: why not once in the beginning of train()

**Review by ofrimasad — COMMENTED** [review]
very nice. but I think we can maybe remove even more parameters. 
my comment inline

File: src/super_gradients/training/sg_trainer/sg_trainer.py (hunk 3 above)
Comment (shaydeci) [reply src/super_gradients/training/sg_trainer/sg_trainer.py:837]: ?

File: src/super_gradients/training/sg_trainer/sg_trainer.py (hunk 1 above)
Comment (shaydeci) [reply src/super_gradients/training/sg_trainer/sg_trainer.py:84]: No, its important that in the constructor we specify whether we use DDP or not, as we initialise DDP in the constructor We do so because of the upcoming objects that would be initialised accordingly (for instance the dataloaders will use an appropriate sampler after doing so).

File: src/super_gradients/training/sg_trainer/sg_trainer.py (hunk 2 above)
Comment (shaydeci) [reply src/super_gradients/training/sg_trainer/sg_trainer.py]: moved to train params in https://github.com/Deci-AI/super-gradients/pull/396/commits/b1d2e101f09d58e6e3374f72aef3d020d73e44ce

File: src/super_gradients/training/sg_trainer/sg_trainer.py (hunk 1 above)
Comment (ofrimasad) [reply src/super_gradients/training/sg_trainer/sg_trainer.py:84]: ok

**Review by ofrimasad — APPROVED** [review]
LGTM
